File: SysAsistenciaAn/src/main/java/pe/edu/upeu/asistencia/models/Usuario.java
package pe.edu.upeu.asistencia.models;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import java.util.HashSet;
import java.util.Set;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity
@Table(name = "gobal_usuario")
public class Usuario {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "nombres", nullable = false)
    @Size(max = 100)
    private String nombres;

    @Column(name = "apellidos", nullable = false)
    @Size(max = 100)
    private String apellidos;

    @Column(nullable = false)
    @Size(max = 100)
    private String correo;

    @Column(nullable = false)
    @Size(max = 100)
    private String password;
    
    @Column(nullable = false)
    @Size(max = 12)
    private String dni;
    
    @Column(name = "perfil_prin", nullable = false)
    @Size(max = 20)
    private String perfilPrin;
    
    @Column(nullable = false)
    @Size(max = 8)
    private String estado;
    
    @Column(nullable = false)
    @Size(max = 2)
    private String offlinex;
    
    @NotNull
    @ManyToMany(fetch = FetchType.EAGER)
    @JoinTable(name = "global_usuario_rol", joinColumns = @JoinColumn(name = "usuario_id"),
            inverseJoinColumns = @JoinColumn(name = "rol_id"))
    private Set<Rol> roles = new HashSet<>();  
    
    @Transient
    private String rolx;
}

File: AsistenciaUPeUJC/app/src/main/java/pe/edu/upeu/asistenciaupeujc/modelo/Usuario.kt
package pe.edu.upeu.asistenciaupeujc.modelo

data class Usuario(
    var nombres: String,
    var apellidos: String,
    var correo: String,
    var password: String,
    var token: String,
    var dni: String,
    var perfilPrin: String,
    var estado: String,
    var offlinex: String,
)

data class UsuarioDto(
    var correo: String,
    var password: String,
)

data class UsuarioResp(
    val id: Long,
    val nombres: String,
    val apellidos: String,
    val correo: String,
    val token: String,
    val dni: String,
    val perfilPrin: String,
    val estado: String,
    val offlinex: String,
)
